src/ut/http_request_test.cpp

```diff
@@ -0,0 +1,201 @@
+/**
+ * @file http_request_test.cpp UT for the HttpRequest interface.
+ *
+ * Copyright (C) Metaswitch Networks 2018
+ * If license terms are provided to you in a COPYING file in the root directory
+ * of the source code repository by which you are accessing this code, then
+ * the license outlined in that COPYING file applies to your use.
+ * Otherwise no rights are granted except for those provided to you by
+ * Metaswitch Networks in a separate written agreement.
+ */
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "mock_httpclient.h"
+#include "http_request.h"
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::DoAll;
+using ::testing::IsEmpty;
+using ::testing::Pointee;
+using ::testing::SetArgReferee;
+using ::testing::SetArgPointee;
+
+
+class HttpRequestTest : public ::testing::Test
+{
+
+  HttpRequestTest()
+  {
+    _client = new MockHttpClient();
+  }
+
+  ~HttpRequestTest()
+  {
+    delete _client; _client = NULL;
+  }
+
+  MockHttpClient* _client;
+};
+
+// Build and send a basic http request, with the default parameters
+// The send method is passed through, so we shouldn't need individual tests to
+// cover all of POST, PUT, GET, DELETE.
+TEST_F(HttpRequestTest, send_basic_request)
+{
+  EXPECT_CALL(*_client, send_request(HttpClient::RequestType::POST, "http://server/testpath", _, _, _, _, _, _, _));
+
+  HttpRequest req = HttpRequest("server", "http", _client, HttpClient::RequestType::POST, "/testpath");
+  req.send();
+}
+
+
+// Check that the HttpRequest defaults non-mandatory arguments to empty/sensible values
+TEST_F(HttpRequestTest, test_default_params)
```

Comment:
It's not obvious to me that yo uneed both this and the above, as they seem to both be testing that `send` is called with `POST` and `http://server/testpath`

Reply:
We could probably remove the above, or wildcard out the method and URL from the below